src/components/molecules/FilterPopover/FilterPopover.tsx

```diff
@@ -0,0 +1,210 @@
+import {useCallback, useState} from 'react';
+import {useDebounce} from 'react-use';
+
+import {Badge, Button, Col, Input, Popover, Row, Space, Tabs} from 'antd';
+
+import {FilterOutlined} from '@ant-design/icons';
+
+import Colors from '@styles/Colors';
+
+import {KeyValue, NewKeyValue} from './KeyValueInput';
+
+export type Filter = {
+  namespace?: string;
+  kind?: string;
+  labels?: Record<string, string>;
+  annotations?: Record<string, string>;
+};
+
+type Props = {
+  filter: Filter | undefined;
+  onChange: (newFilter: Filter | undefined) => void;
+};
+
+export function FilterPopover({filter, onChange}: Props) {
+  const filterCount = filter ? Object.keys(filter).length : 0;
+  const [namespace, setNamespace] = useState<string | undefined>(filter?.namespace);
+  const [kind, setKind] = useState<string | undefined>(filter?.kind);
+
+  const handleChange = useCallback(
+    (newFilter: Filter) => {
+      const isEmpty = Object.keys(newFilter).length === 0;
+      onChange(isEmpty ? undefined : newFilter);
+    },
+    [onChange]
+  );
+
+  useDebounce(
+    () => {
+      if (namespace === filter?.namespace) return;
+      const newFilter = {...filter, namespace};
+      if (!namespace) delete newFilter.namespace;
```

Comment:
It is because you get `{ namespace: undefined }` while I want the resulting filter to be as minimal as possible. I also count filters with `Object.keys(filter).length` so unless I compare `undefined` values the alternative would give subtle bugs.